### PR TITLE
custom DatePicker component

### PR DESCRIPTION
**Pull request by bryce-fitzsimons**
image

image

image

### CODE REVIEW

File: packages/front-end/components/Events/EventsPage/EventsPage.tsx [hunk 1]
```diff
@@ -224,44 +228,38 @@ export const EventsPageContainer = () => {
...
           label="To"
-          className="text-muted"
-          labelClassName="mr-2 mb-0"
-          containerClassName="ml-2 d-flex align-items-center mb-0"
-          value={toDate ? toDate.toISOString().split("T")[0] : ""}
-          onChange={(d) => {
-            setToDate(d.target.value ? new Date(d.target.value) : null);
-          }}
+          precision="date"
+          containerClassName=""
```
Comment (jdorn) [inline packages/front-end/components/Events/EventsPage/EventsPage.tsx:247]: Why not use the date range functionality here instead of 2 inputs?

**Review by jdorn — APPROVED** [review]

File: packages/front-end/components/Events/EventsPage/EventsPage.tsx (hunk 1 above)
Comment (bryce-fitzsimons) [reply packages/front-end/components/Events/EventsPage/EventsPage.tsx:247]: The interval start/end dates are both optional, which doesn't play nicely with this widget. We could eventually add custom support for optional ranges

**Review by ahdriel — COMMENTED** [review]
Clicking around I was able to make my end date be before my start date. Is this something we want to allow for a daterange or would it be better to say that a range always has the earlier date on the left?

Screenshot 2024-11-14 at 8 57 06 AM

File: packages/front-end/test/services/async-queue.test.ts [hunk 2]
```diff
@@ -1,3 +1,4 @@
+import { vi } from "vitest";
```
[the remark this answers is not in the document]
Comment (bryce-fitzsimons) [reply packages/front-end/test/services/async-queue.test.ts:1]: jest was unable to compile some of the ESM modules in the libraries that were updated (date-fns namely) as part of this change. it's a known issue with jest. vitest fixes it and is generally a better test engine.

**Pull request merged**